packages/cli-kit/src/store.ts

```diff
@@ -183,4 +193,18 @@ export class CLIKitStore extends Conf<ConfSchema> {
     debug(content`Removing session store...`)
     this.set('sessionStore', '')
   }
+
+  async cache<TValue>(key: string, source: () => Promise<TValue>, expirySeconds = 24 * 3600): Promise<TValue> {
+    const expiryMilliseconds = expirySeconds * 1000
+    const cache = this.get('cache') || {}
+    if (cache[key]) {
```

Comment:
if you pull `cache[key]` into a variable before hitting the `if`, TS's type narrowing should mean you need a lot fewer `!`.